pkg/logql/log/parser.go

```diff
@@ -500,7 +500,7 @@ func (j *JSONExpressionParser) Process(_ int64, line []byte, lbs *LabelsBuilder)
 	// Check that the line starts correctly
 	// the parser will pass an error if other
 	// parts of the line are malformed
-	if !isValidJSONStart(line) {
+	if len(line) == 0 || !isValidJSONStart(line) {
```

Comment:
I think we should handle this case in its own branch; there's no JSON validation error here, strictly speaking.
I think it can just fail silently.